packages/gatsby/src/commands/build.js

```diff
@@ -201,4 +249,52 @@ module.exports = async function build(program: BuildArgs) {
   await stopTracer()
   workerPool.end()
   buildActivity.end()
+
+  if (
+    process.env.GATSBY_EXPERIMENTAL_PAGE_BUILD_ON_DATA_CHANGES &&
+    process.argv.includes(`--log-pages`)
+  ) {
+    if (pagePaths.length) {
+      report.info(
+        `Built pages:\n${pagePaths
+          .map(path => `Updated page: ${path}`)
+          .join(`\n`)}`
+      )
+    }
+
+    if (deletedPageKeys.length) {
+      report.info(
+        `Deleted pages:\n${deletedPageKeys
+          .map(path => `Deleted page: ${path}`)
+          .join(`\n`)}`
+      )
+    }
+  }
+
+  if (
+    process.env.GATSBY_EXPERIMENTAL_PAGE_BUILD_ON_DATA_CHANGES &&
+    process.argv.includes(`--write-to-file`)
+  ) {
+    const createdFilesPath = path.resolve(
+      `${program.directory}/.cache`,
+      `newPages.txt`
+    )
+    const deletedFilesPath = path.resolve(
+      `${program.directory}/.cache`,
+      `deletedPages.txt`
+    )
+
+    if (pagePaths.length) {
+      await fs.writeFile(createdFilesPath, `${pagePaths.join(`\n`)}\n`, `utf8`)
+      report.info(`newPages.txt created`)
```

Comment:
add subdir to the info?

```suggestion
      report.info(`.cache/newPages.txt created`)
```

Reply:
Added